src/NodeAnalyzer/CompactFuncCallAnalyzer.php

```diff
@@ -27,13 +28,32 @@ public function isInCompact(FuncCall $funcCall, Variable $variable): bool
             return false;
         }
 
-        $args = $funcCall->args;
-        foreach ($args as $arg) {
-            if (! $arg->value instanceof String_) {
+        return $this->isInArgOrArrayItemNodes($funcCall->args, $variableName);
+    }
+
+    /**
+     * @param Array<\PhpParser\Node\Arg|\PhpParser\Node\Expr\ArrayItem|null> $nodes
```

Comment:
Can be updated to : 

```suggestion
     * @param array<int, \PhpParser\Node\Arg|\PhpParser\Node\Expr\ArrayItem|null> $nodes
```

for consistency with other `@param` usage.